### PR TITLE
[TE] endpoint for anomaly time ranges per metric id

### PR DESCRIPTION
**Pull request by apucher**
* MergedAnomaly DAO method to fetch anomalies for a metricId (via anomaly function index) within a time range

* MergedAnomaly DAO method to fetch anomalies for a list of metricIds (via anomaly function index) within a time range

* Endpoint to fetch map of time ranges (keyed by metricId) for a list of metricIds within a time range

### CODE REVIEW

File: thirdeye/thirdeye-pinot/src/main/java/com/linkedin/thirdeye/dashboard/resources/v2/DataResource.java [hunk 1]
```diff
@@ -543,6 +547,81 @@ public WowSummary getWowSummary(
...
   }
 
+  @GET
+  @Path("anomalies/ranges")
+  public Map<Long, List<TimeRange>> getAnomalyTimeRangesByMetricIds(
+      @QueryParam("metricIds") String metricIds,
+      @QueryParam("start") Long start,
+      @QueryParam("end") Long end) {
+
+    if (metricIds == null)
```
Comment (npawar) [inline thirdeye/thirdeye-pinot/src/main/java/com/linkedin/thirdeye/dashboard/resources/v2/DataResource.java]: Is blank metricIds okay?

File: thirdeye/thirdeye-pinot/src/main/java/com/linkedin/thirdeye/dashboard/resources/v2/DataResource.java [hunk 2]
```diff
@@ -543,6 +547,81 @@ public WowSummary getWowSummary(
...
+      throw new IllegalArgumentException("Must provide metricIds");
+
+    if (start == null)
+      throw new IllegalArgumentException("Must provide start timestamp");
+
+    if (end == null)
```
Comment (npawar) [inline thirdeye/thirdeye-pinot/src/main/java/com/linkedin/thirdeye/dashboard/resources/v2/DataResource.java]: Why not use path params if they're all mandatory inputs?

File: thirdeye/thirdeye-pinot/src/main/java/com/linkedin/thirdeye/dashboard/resources/v2/DataResource.java [hunk 3]
```diff
@@ -543,6 +547,81 @@ public WowSummary getWowSummary(
...
+
+    Map<Long, List<TimeRange>> output = new HashMap<>();
+    for(Map.Entry<Long, List<MergedAnomalyResultDTO>> entry : anomalies.entrySet()) {
+      output.put(entry.getKey(), truncateRanges(extractAnomalyTimeRanges(entry.getValue()), start, end));
+    }
+
+    return output;
+  }
+
+  static List<TimeRange> extractAnomalyTimeRanges(List<MergedAnomalyResultDTO> anomalies) {
```
Comment (npawar) [inline thirdeye/thirdeye-pinot/src/main/java/com/linkedin/thirdeye/dashboard/resources/v2/DataResource.java]: javadocs for these two (extractAnomalyTimeRanges  and truncateRanges) would be helpful

File: thirdeye/thirdeye-pinot/src/main/java/com/linkedin/thirdeye/dashboard/resources/v2/DataResource.java (hunk 1 above)
Comment (apucher) [inline thirdeye/thirdeye-pinot/src/main/java/com/linkedin/thirdeye/dashboard/resources/v2/DataResource.java]: technically, yes. If merticIds is empty it just returns an empty map

File: thirdeye/thirdeye-pinot/src/main/java/com/linkedin/thirdeye/dashboard/resources/v2/DataResource.java (hunk 2 above)
Comment (apucher) [inline thirdeye/thirdeye-pinot/src/main/java/com/linkedin/thirdeye/dashboard/resources/v2/DataResource.java]: Stylistic question. Absolutely doable, yet start/stop aren't really properties of the anomaly ranges either (i.e. they're a cutoffs, not anomaly ranges)

File: thirdeye/thirdeye-pinot/src/main/java/com/linkedin/thirdeye/dashboard/resources/v2/DataResource.java (hunk 3 above)
Comment (apucher) [inline thirdeye/thirdeye-pinot/src/main/java/com/linkedin/thirdeye/dashboard/resources/v2/DataResource.java]: fixed

**Pull request merged**